homeassistant/components/xiaomi_ble/config_flow.py

```diff
@@ -53,6 +72,12 @@ async def async_step_bluetooth(
         device = DeviceData()
         if not device.supported(discovery_info):
             return self.async_abort(reason="not_supported")
+
+        # Wait until we have received enough information about this device to detect its encryption type
+        await self._async_wait_for_full_advertisement(discovery_info, device)
```

Comment:
We need to catch `asyncio.TimeoutError` here, right?

Reply:
Probably! I'm just working on tests now to make sure i'm happy with the API of the helpers. I think I will let them raise, and then abort the config flow on the TimeoutError, rather than checking device.pending again (If there is a Timeout it will still be pending, if there is not then it bailed early and hence is not pending).

Reply:
Latest push does this and has tests for both step_user and step_bluetooth where the timeout expires and also where a suitable advertisement is injected, allowing the flow to recover and correctly detect the encryption type.